lightningd/pay.c

```diff
@@ -630,6 +630,24 @@ static bool should_use_tlv(enum route_hop_style style)
 	abort();
 }
 
+static struct htlc_out *send_onion(struct lightningd *ld,
+				   const struct onionpacket *packet,
+				   const struct route_hop *first_hop,
+				   const struct sha256 *payment_hash,
+				   struct channel *channel,
+				   enum onion_type *failcode)
+{
+	const u8 *onion;
+	unsigned int base_expiry;
+	struct htlc_out *hout;
+	base_expiry = get_block_height(ld->topology) + 1;
+	onion = serialize_onionpacket(tmpctx, packet);
+	*failcode = send_htlc_out(channel, first_hop->amount,
+				  base_expiry + first_hop->delay,
+				  payment_hash, onion, NULL, &hout);
```

Comment:
The inversion of failcode and hout here gives me whiplash.

"Always return the error"!

Reply:
I see, I guess `send_onion` should follow that convention as well :-)